feat: Add configurable timeout seconds setting, move to settings.ts

slaughtr opened the pull request:
This makes the timeout delay for debounce configurable.

* Moved settings to `settings.ts`
* Created settings `display()` method
* Created setting for "Preview Mode Switch Delay"

I wanted to enforce the entering of actual numbers in the settings but seems there's no real validation.

Commit:
feat: Add configurable timeout seconds setting, move to settings.ts

OwnJoke commented:
Thanks for this, ill probably check everything and merge it, as well as revert some of the code back a version so the plugin actually works this weekend.

OwnJoke committed:
fix the switching not working

this brings back issues where certain text areas that arent the document force you to start typing in the document but at least it works

OwnJoke commented:
Hey there, I updated your main.ts to actually make it so the switching works but the setting does not seem to adjust the time, this could be me that messed something up while updating the file but i'm not sure and I can't seem to figure out what's wrong.

slaughtr commented:
Didn't get a chance to look at this over the weekend. Nothing jumps out in your changes, but I'll try to test this week.

FWIW I've been running my fork for almost 2 weeks without any issues. I use `cmd+e` ("Toggle edit/preview mode"), which seems to eliminate any issues with switching. That could be part of my luck.

OwnJoke commented:
Live preview is now out so it doesn't really matter. I'll look into it anyway but this plug-in is pretty much useless now.

slaughtr commented:
Unfortunately I have to agree. Would be nice to have the plugin working but personally don't think I'll find the time, especially now that live preview is there and working.
